### PR TITLE
Adding postgresql firewall rule facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY

Adding PostgreSQL firewall rule facts module

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
azure_rm_postgresqlfirewallrule_facts

##### ANSIBLE VERSION
2.7

### CODE REVIEW

**Comment by Fred-sun** [comment]
@zikalino

**Comment by Fred-sun** [comment]
@yuwzho @yungezz

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_postgresqlfirewallrule_facts
+version_added: "2.8"
+short_description: Get Azure PostgreSQL Firewall Rule facts.
+description:
+    - Get facts of PostgreSQL Firewall Rule.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py]: Azure

File: test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml [hunk 2]
```diff
@@ -300,6 +300,41 @@
...
+- name: Gather facts PostgreSQL Firewall Rule
+  azure_rm_postgresqlfirewallrule_facts:
+    resource_group: "{{ resource_group }}"
+    server_name: postgresqlsrv{{ rpfx }}
+    name: firewallrule{{ rpfx }}
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.changed == False
```
Comment (yungezz) [inline test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml:320]: length check?

File: test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml [hunk 3]
```diff
@@ -300,6 +300,41 @@
...
+- name: Gather facts PostgreSQL Firewall Rule
+  azure_rm_postgresqlfirewallrule_facts:
+    resource_group: "{{ resource_group }}"
+    server_name: postgresqlsrv{{ rpfx }}
+  register: output
+- name: Assert that facts are returned
+  assert:
+    that:
+      - output.changed == False
+      - output.rules[0].id != None
```
Comment (yungezz) [inline test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml:337]: is this solid enough? always returned in this order?

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+      server_name: server_name
+'''
+
+RETURN = '''
+rules:
+    description: A list of dictionaries containing facts for PostgreSQL Firewall Rule.
+    returned: always
+    type: complex
+    contains:
+        id:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py:64]: even no postsql server information?

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py]: ok fixed

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py (hunk 4 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py:64]: yes, that's fixed now

File: test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml (hunk 2 above)
Comment (zikalino) [reply test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml:320]: done

File: test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml (hunk 3 above)
Comment (zikalino) [reply test/integration/targets/azure_rm_postgresqlserver/tasks/main.yml:337]: hmm... yes if I understand correctly

File: lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            'server_name': self.server_name,
+            'name': d['name'],
+            'start_ip_address': d['start_ip_address'],
+            'end_ip_address': d['end_ip_address']
+        }
+        return d
+
+
+def main():
+    AzureRMFirewallRulesFacts()
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_postgresqlfirewallrule_facts.py]: I think there're lots of firewalls for different dbs, pls naming the class specifically.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
shipit

**Pull request merged**